### PR TITLE
Payment Button Block: InnerBlocks have the ability to connect Stripe

### PR DESCRIPTION
**Pull request by rcrdortiz**
Now the InnerBlocks of the Payment Button block is able to connect Stripe.

Fixes [#61422](https://github.com/Automattic/wp-calypso/issues/61422)

#### Changes proposed in this Pull Request:

* Propagated the block controls to it's children using the `__experimentalExposeControlsToChildren` Gutenberg supports attribute.

#### Known issues
The `__experimentalExposeControlsToChildren` does not support choosing a subset of controls to inherit. In order to avoid having alignment controls twice on the child blocks, I've removed support for aligning on the Payment Button block. We can still align the buttons as we did before, the only difference will be that the main block won't have this option. I don't anticipate this being a blocker since the component maintains all its previous functionality. If someone has a strong opinion, we should tackle it on a separate issue.

I found a bug that is also present in the previous version. When working with the block, if you set an alignment you can then bypass the template locking and insert blocks in the button row. I've created an [issue to address this problem](https://github.com/Automattic/jetpack/issues/23204).

#### Testing instructions:

* Edit or create a new Post.
* Add a Payment Button block.
* Make sure you have Stripe disconnected.
* Click on the block, you should see the `Connect Stripe` option in the toolbar.
* Click on the button inside the block, you should see the `Connect Stripe` option in the toolbar.

#### Test evidence

https://user-images.githubusercontent.com/1989914/156445501-b38edb0a-abf6-4f04-aca7-2e712fe64cad.mp4

### CODE REVIEW

**Review by jeherve — DISMISSED** [review]
You'll need to fix the failing tests before we can merge this:
https://github.com/Automattic/jetpack/runs/5398008967?check_suite_focus=true

File: projects/plugins/jetpack/extensions/blocks/recurring-payments/edit.jsx [hunk 1]
```diff
@@ -478,7 +478,7 @@ export class MembershipsButtonEdit extends Component {
 		);
 
 		const blockControls = (
-			<BlockControls>
+			<BlockControls __experimentalShareWithChildBlocks>
```
[the remark this answers is not in the document]
Comment (rcrdortiz) [reply projects/plugins/jetpack/extensions/blocks/recurring-payments/edit.jsx:481]: I'll test it and get back to you.

[the remark this answers is not in the document]
Comment (rcrdortiz) [reply projects/plugins/jetpack/extensions/blocks/recurring-payments/edit.jsx:481]: Tested on an Atomic site WordPress 5.8.2, works well.

**Review by BogdanUngureanu — APPROVED** [review]
Works like a charm on both Atomic (with and without Gutenberg plugin enabled) and on WPCOM. 🎉 

I think we should enable the same behaviour on Premium content block too, where this would be a lot more helpful since there are a lot more nested blocks (either in this PR or in a separate one).

**Review by samiff — APPROVED** [review]
Tested well for me 👍

**Comment by samiff** [comment]
r241299-wpcom

**Pull request merged**